src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -121,20 +121,20 @@ public User getUser(String email) {
   {
     Set<String> users = new HashSet<>();
 	
-    /*adding users who have posted a message*/
-    Query query = new Query("Message");
-    PreparedQuery results = datastore.prepare(query);
-    for(Entity entity: results.asIterable())
-    {
-      users.add((String)entity.getProperty("user"));
-    }
-	
     /*adding users who have added about me*/
     Query query2 = new Query("User");
     PreparedQuery results2 = datastore.prepare(query2);
     for(Entity entity: results2.asIterable())
     {
-      users.add((String)entity.getProperty("email"));
+      users.add((String)entity.getProperty("email") + (String)entity.getProperty("aboutMe"));
```

Comment:
Instead of returning a string, have you considering creating a User object and returning that in the response? Then in community-page-loader.js, you wouldn't have to rely on string parsing to get the proper values. Take a look at how Datastore#getMessages() and MessageServlet#doGet() if you like.

Otherwise, if you prefer to leave it as a string, then I'd suggest adding some sort of dividing character(s) instead of  relying on ".com", since emails are not guaranteed to have this domain name (ie. could have .ca, .edu). However, that could also run into some issues if that set of characters if found in the "About me" text (especially since the response is sent in plaintext).

Reply:
Also, how does adding "email"+"aboutMe" properties different from adding the "user" property (line 137)? These should probably be of the same format, since they are being treated the same on the client.

Reply:
I believe user property in "Message" is same as email property in "User", so it is like I am either adding userX + aboutMe or userX to the HashSet, but the problem that I missed is that they hash to different things . I thought of returning set of objects, but then I though meybe I should not change the function interface. However, it seems like a better option.

Reply:
One problem with returning set of user objects is that I don't know how to get information out of it in the JavaScript file. I am just learning javascript and Json, but I can try

Reply:
Aside, feel free to change the method signature to something easier to work with - I don't think it is used anywhere else :)